odl/operator/operator.py

```diff
@@ -245,6 +518,74 @@ def __init__(self, domain, range, linear=False):
                 raise TypeError('range {!r} not a `LinearSpace` or `Field` '
                                 'instance.'.format(range))
 
+    def _call(self, x, out=None, **kwargs):
+        """Implementation of the operator evaluation.
+
+        This method is private backend for the evaluation of this
+        operator. It needs to match certain signature conventions,
+        and its implementation type is inferred from its signature.
+
+        The following signatures are allowed:
+
+        Python 2 and 3:
+            - ``_call(self, x)``  -->  out-of-place evaluation
+            - ``_call(self, vec, out)``  -->  in-place evaluation
+            - ``_call(self, x, out=None)``   --> both
+
+        Python 3 only:
+            - ``_call(self, x, *, out=None)`` (``out`` as keyword-only
+              argument)  --> both
+
+        For disambiguation, the instance name (the first argument) **must**
+        be 'self'.
+
+        The name of the ``out`` argument **must** be 'out', the second
+        argument may have any name.
+
+        Additional variable ``**kwargs`` and keyword-only arguments
+        (Python 3 only) are also allowed.
+
+        **General advice:**
+        - We recommend to implement the *in-place*
```

Comment:
This may be "advanced" or something like that, mention that it is for performance reasons.